server/src/main/java/org/elasticsearch/threadpool/ThreadPool.java

```diff
@@ -239,16 +251,7 @@ public Counter estimatedTimeInMillisCounter() {
     }
 
     public ThreadPoolInfo info() {
-        List<Info> infos = new ArrayList<>();
-        for (ExecutorHolder holder : executors.values()) {
-            String name = holder.info.getName();
-            // no need to have info on "same" thread pool
-            if ("same".equals(name)) {
-                continue;
-            }
-            infos.add(holder.info);
-        }
-        return new ThreadPoolInfo(infos);
```

Comment:
There is no need to recreate this info (garbage!) on every invocation, it is immutable.